public/app/features/alerting/unified/hooks/useAbilities.ts

```diff
@@ -202,7 +295,30 @@ export function useAlertmanagerAbilities(actions: AlertmanagerAction[]): Ability
   }, [abilities, actions]);
 }
 
-export function useAlertSourceAbility(action: AlertSourceAction): Ability {
-  const abilities = useAlertSourceAbilities();
-  return useMemo(() => abilities[action], [abilities, action]);
+/**
+ * We don't want to show the silence button if either
+ * 1. the user has no permissions to create silences
+ * 2. the admin has configured to only send instances to external AMs
+ */
+export function useCanSilence(rulesSource: RulesSource): [boolean, boolean] {
```

Comment:
what about making this hook private to avoid using it directly ?

Reply:
Definitely, this hook shouldn't be called directly but use `useAlertRuleAbility(rule, AlertRuleAction.Silence)` instead